lib/url.js

```diff
@@ -51,7 +51,9 @@ const nonHostChars = ['%', '/', '?', ';', '#'].concat(autoEscape);
 const hostEndingChars = ['/', '?', '#'];
 const hostnameMaxLen = 255;
 const hostnamePartPattern = /^[+a-z0-9A-Z_-]{0,63}$/;
-const hostnamePartStart = /^([+a-z0-9A-Z_-]{0,63})(.*)$/;
+const hostnamePartStart = /^([^:\/?#\t\r\n]{0,63})(.*)$/;
+const hostnameTerminators = /[:\/?#\t\r\n]/;
```

Comment:
@thefourtheye OK, I made that change as well as the change to use `\w` character class. For consistency, I also changed two other instances in the same file to which those comments would apply. Will re-run CI momentarily...